theme/aplus/static/default.css

```diff
@@ -97,467 +108,528 @@ dt:hover > a.headerlink,
 }
 
 .viewcode-back {
-    float: right;
-    font-family: sans-serif;
+  float: right;
+  font-family: sans-serif;
 }
 
 div.viewcode-block:target {
-    margin: -1px -10px;
-    padding: 0 10px;
+  margin: -1px -10px;
+  padding: 0 10px;
 }
 
 /* -- math display ---------------------------------------------------------- */
 
 img.math {
-    vertical-align: middle;
+  vertical-align: middle;
 }
 
 div.body div.math p {
-    text-align: center;
+  text-align: center;
 }
 
 span.eqno {
-    float: right;
+  float: right;
 }
 
 span.eqno a.headerlink {
-    position: relative;
-    left: 0px;
-    z-index: 1;
+  position: relative;
+  left: 0px;
+  z-index: 1;
 }
 
 div.math:hover a.headerlink {
-    visibility: visible;
+  visibility: visible;
 }
 
 /* -- printout stylesheet --------------------------------------------------- */
 
 @media print {
-    div.document,
-    div.documentwrapper,
-    div.bodywrapper {
-        margin: 0 !important;
-        width: 100%;
-    }
+  div.document,
+  div.documentwrapper,
+  div.bodywrapper {
+    margin: 0 !important;
+    width: 100%;
+  }
 
-    div.sphinxsidebar,
-    div.related,
-    div.footer,
-    #top-link {
-        display: none;
-    }
+  div.sphinxsidebar,
+  div.related,
+  div.footer,
+  #top-link {
+    display: none;
+  }
 }
 
 /* -- point of interest --------------------------------------------------- */
 
 .poi {
-    margin: 20px 0;
-    padding: 0;
-    width:inherit;
+  margin: 20px 0;
+  padding: 0;
+  width: inherit;
 }
 
 .no-poi-box {
-    margin: 0;
-    padding: 0;
-    width: inherit;
+  margin: 0;
+  padding: 0;
+  width: inherit;
 }
 
 .poi-container {
-    width:100%;
-    padding:0;
-    margin: 5px 0;
+  width: 100%;
+  padding: 0;
+  margin: 5px 0;
 }
 
 .poi-content {
-    border: 1px solid #ccc;
-    padding:15px 20px;
-    border-radius:4px;
-}   
+  border: 1px solid #ccc;
+  padding: 15px 20px;
+  border-radius: 4px;
+}
 
 .poi-title {
-    width: 80%;
-    float:left;
-    height:20px;
-    font-weight: bold;
-    font-size:1.2em;
-    padding-left:0;
-} 
+  width: 80%;
+  float: left;
+  height: 20px;
+  font-weight: bold;
+  font-size: 1.2em;
+  padding-left: 0;
+}
 
 .poi-links {
-    width:20%;
-    text-align:right;
-    float:left;
-    margin: 5px 0 0 0;
-    height:20px;
+  width: 20%;
+  text-align: right;
+  float: left;
+  margin: 5px 0 0 0;
+  height: 20px;
 }
 
 .poi-icon {
-    margin: 0 10px;
-    vertical-align: baseline;
-    height:20px;
+  margin: 0 10px;
+  vertical-align: baseline;
+  height: 20px;
 }
 
 .borderless .poi-content {
-    border: 0;
+  border: 0;
 }
 
 .poi-container.poi-content {
-    border: 1px solid #e7f2fa;
-    border-top: 2px solid #3c763d;
-    border-bottom: 2px solid #3c763d;
-    background: #ebf5e7;
+  border: 1px solid #e7f2fa;
+  border-top: 2px solid #3c763d;
+  border-bottom: 2px solid #3c763d;
+  background: #ebf5e7;
+}
+
+/* The current version of A+ use Bootstrap 3. The following classes will emulate 
```

Comment:
Should this be actually implemented in A+? After all, they edit the non course context (i.e area out of `#course-content`). To be fair, all css here should be written so that it only affects content below course area and not the course menu for example.

Reply:
It is true, but if we talk about sphinx themes, those themes are intended to change the whole UI of the documentation. So I am confused now. I am not sure if the intention of having the aplus-theme here is to change the whole UI as the sphinx themes do, or the aplus-theme is only for styling a-plus-rst-tools elements. In that case, this is not a theme in the whole sense of the word. Could you clarify this idea a bit better?

Reply:
We don't use Sphinx in the conventional way since we are embedding those HTML pages in another system. I don't think our default style in the a-plus-rst-tools should affect the A+ platform itself, like the menus. A-plus-rst-tools should focus on the course contents. If some course has styles that affect the A+ menus too, they can do that, but it is not the default behaviour. If we want to change the styles of the whole platform for everyone, we can do it in A+, not in a-plus-rst-tools.

Reply:
The course specific CSS shouldn't ever edit A+ frame, only the course content area. In addition, they shouldn't edit A+ elements, like exercise frames, by default. Basically, if we fix some A+ outlooks in the rst-tools, then courses which don't use that would still use the basic styles from A+, which at least makes the courses look different.

I think the "aplus theme" was originally done to make the static html files look-a-like of A+ installation, as it was part of the development workflow. Now that the local files shouldn't normally be checked, we could easily strip away all look-a-like parts (like top and side navigation). Thus, the concept of theme would become simpler.

However, themes would be nice way to support thinks like O1 material, where the material is very different from the default stuff.